flocker/control/httpapi.py

```diff
@@ -351,7 +319,7 @@ def saved(ignored):
                       '/v1/endpoints.json#/definitions/configuration_dataset'},
         schema_store=SCHEMAS
     )
-    def update_dataset(self, dataset_id, primary=None):
+    def update_dataset(self, dataset_id, primary=None, maximum_size=None):
```

Comment:
This is not good. Our model currently uses `maximum_size=None` to indicate "no particular size set". This is not relevant to block storage backends, but supported by ZFS. And the only reasonable way to expose the JSON API for that is likely `{"maximum_size": null}`. I guess technically one could do `DELETE /v1/configuration/datasets/21321412/maximum_size` but that seems a little weird.

Assuming we don't go with `DELETE`, we now have two meanings for `None`: parameter not passed, and "remove maximum size requirement."

Instead you should have a `_NO_ARGUMENT = object()` on module level and then default value of `maximum_size` can be that, allowing you to distinguish the two cases.

You should then implement support for setting `maximum_size` to None.